src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/DiscoverVariantsFromContigAlignmentsSAMSpark.java

```diff
@@ -167,21 +180,119 @@ public static AlignedContig parseReadsAndOptionallySplitGappedAlignments(final I
                         .mapToPair(alignedContig -> new Tuple2<>(alignedContig.contigSequence,                                        // filter a contig's alignment and massage into ordered collection of chimeric alignments
                                 ChimericAlignment.parseOneContig(alignedContig, DEFAULT_MIN_ALIGNMENT_LENGTH)))
                         .flatMapToPair(DiscoverVariantsFromContigAlignmentsSAMSpark::discoverNovelAdjacencyFromChimericAlignments)    // a filter-passing contig's alignments may or may not produce novel adjacency
                         .groupByKey()                                                                                                 // group the same novel adjacency produced by different contigs together
                         .mapToPair(noveltyAndEvidence -> inferType(noveltyAndEvidence._1, noveltyAndEvidence._2))                     // type inference based on novel adjacency and evidence alignments
-                        .map(noveltyTypeAndEvidence -> annotateVariant(noveltyTypeAndEvidence._1,                                     // annotate the novel adjacency and inferred type
-                                noveltyTypeAndEvidence._2._1, noveltyTypeAndEvidence._2._2, broadcastReference));
+            .map(noveltyTypeAndEvidence -> annotateVariant(noveltyTypeAndEvidence._1,                                     // annotate the novel adjacency and inferred type
+                                                           noveltyTypeAndEvidence._2._1, noveltyTypeAndEvidence._2._2,
+                    broadcastReference));
+
+        List<VariantContext> collectedAnnotatedVariants = annotatedVariants.collect();
+
+        if (evidenceTargetLinks != null) {
+            collectedAnnotatedVariants = processEvidenceTargetLinks(parameters,
+                    localLogger,
+                    evidenceTargetLinks,
+                    metadata,
+                    collectedAnnotatedVariants, broadcastReference.getValue());
+        }
+
+        SVVCFWriter.writeVCF(vcfOutputFileName, localLogger, collectedAnnotatedVariants,
+                sequenceDictionary);
+    }
+
+    @VisibleForTesting
+    static List<VariantContext> processEvidenceTargetLinks(final StructuralVariationDiscoveryArgumentCollection.DiscoverVariantsFromContigsAlignmentsSparkArgumentCollection parameters,
+                                                                   final Logger localLogger,
+                                                                   final PairedStrandedIntervalTree<EvidenceTargetLink> evidenceTargetLinks,
+                                                                   final ReadMetadata metadata,
+                                                                   List<VariantContext> collectedAnnotatedVariants,
```

Comment:
Since we need to first compare the evidence links to the variants, and then the variants to the evidence target links, we'd either need to first broadcast one and then the other, or else come up with some sort of joining scheme. This seems much more efficient to just collect them all into local collections and compare on one machine.